node/core/approval-voting/src/criteria.rs

```diff
@@ -88,25 +91,105 @@ impl From<OurAssignment> for crate::approval_db::v1::OurAssignment {
 	}
 }
 
-fn relay_vrf_modulo_transcript(relay_vrf_story: RelayVRFStory, sample: u32) -> Transcript {
-	// combine the relay VRF story with a sample number.
-	let mut t = Transcript::new(approval_types::RELAY_VRF_MODULO_CONTEXT);
-	t.append_message(b"RC-VRF", &relay_vrf_story.0);
-	sample.using_encoded(|s| t.append_message(b"sample", s));
+// Combines the relay VRF story with a sample number if any.
+fn relay_vrf_modulo_transcript_inner(
+	mut transcript: Transcript,
+	relay_vrf_story: RelayVRFStory,
+	sample: Option<u32>,
+) -> Transcript {
+	transcript.append_message(b"RC-VRF", &relay_vrf_story.0);
 
-	t
+	if let Some(sample) = sample {
+		sample.using_encoded(|s| transcript.append_message(b"sample", s));
+	}
+
+	transcript
+}
+
+fn relay_vrf_modulo_transcript_v1(relay_vrf_story: RelayVRFStory, sample: u32) -> Transcript {
+	relay_vrf_modulo_transcript_inner(
+		Transcript::new(approval_types::v1::RELAY_VRF_MODULO_CONTEXT),
+		relay_vrf_story,
+		Some(sample),
+	)
+}
+
+fn relay_vrf_modulo_transcript_v2(relay_vrf_story: RelayVRFStory) -> Transcript {
+	relay_vrf_modulo_transcript_inner(
+		Transcript::new(approval_types::v2::RELAY_VRF_MODULO_CONTEXT),
+		relay_vrf_story,
+		None,
+	)
+}
+
+/// A hard upper bound on num_cores * target_checkers / num_validators
+const MAX_MODULO_SAMPLES: usize = 40;
```

Comment:
Is it really necessary to hardcode this?

Reply:
for now it is required by `struct BigArray(pub [u8; MAX_MODULO_SAMPLES * 4]);`